common-http/src/test/java/io/cdap/common/http/HttpRequestsTestBase.java

```diff
@@ -110,13 +127,39 @@ public void testHttpStatus() throws Exception {
     testDelete("/api/testDelete", only(200), any(), any(), any());
   }
 
+  private HttpRequest getRequest(HttpRequest.Builder builder) {
+    if (returnResponseStream()) {
+      // Set a small chunk size to verify that we're doing chunked reads correctly.
+      int chunkSize = 5;
+      return builder.withContentConsumer(new HttpContentConsumer(chunkSize) {
+        @Override
+        public boolean onReceived(ByteBuffer buffer) {
+          // create byte array with length = number of bytes written to the buffer
+          byte[] bytes = new byte[buffer.remaining()];
+          // read the bytes that were written to the buffer
+          buffer.get(bytes);
+          streamingResponseBody = streamingResponseBody + new String(bytes);
```

Comment:
Always pass in codec when creating string from byte array. `new String(bytes, StandardCharsets.UTF_8)`

Reply:
fixed